dev/tools/gen_defaults/lib/snackbar_template.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'template.dart';
+
+class SnackbarTemplate extends TokenTemplate {
+  const SnackbarTemplate(
+      this.tokenGroup, super.blockName, super.fileName, super.tokens);
+
+  final String tokenGroup;
+
+  @override
+  String generate() => '''
+class _${blockName}DefaultsM3 extends SnackBarThemeData {
+  const _${blockName}DefaultsM3(this.context);
+
+  final BuildContext context;
```

Comment:
We can make this a little more efficient if we cache the Theme.of(context) calls below by adding these late variables:

```dart
late final ThemeData _theme = Theme.of(context);
late final ColorScheme _colors = _theme.colorScheme;
```

And then in the constructor set the color prefix to:

```dart
  const SnackbarTemplate(super.blockName, super.fileName, super.tokens, {
    super.colorSchemePrefix = '_colors.'
  });
```

You can also add a similar one for `textTheme` if you are doing a lot of text theme look ups.

Reply:
Done, though I have to remove the `cosnt` constructor of SnackBarDefaultsM3. I don't think there's a way around that?

Reply:
I don't think so, but there is never going to be a const with a `BuildContext` here anyway, so I think the trade off is worth it. Thx.